### PR TITLE
Data loader provide parent ref to observe scroll trm 24703

### PR DESCRIPTION
**Pull request by dlrice**
Relevant jira: https://www.ebi.ac.uk/panda/jira/browse/TRM-24703

Required refactoring the data loader to accept a reference (eg found with a query selector) as a parent in the uniprot website requires observing rather than a child.

### CODE REVIEW

File: src/components/data-loader.jsx [hunk 1]
```diff
@@ -36,28 +52,33 @@ const withDataLoader = BaseComponent => (props) => {
...
-  };
-
-  const checkLoadMoreItems = () => {
-    if (!loading && hasMoreData && isBottom(ref)) {
-      setLoadMoreItems(true);
-    }
-  };
-
+  useEffect(() => {
+    const ref = scrollRef.current;
```
Comment (xwatkins) [inline src/components/data-loader.jsx:56]: Wondering whether you need to check if `ref` is not null before continuing?

File: src/styles/components/data-table.scss [hunk 2]
```diff
@@ -26,7 +26,7 @@
     color: $colour-sky-white;
     background-color: $colour-sea-blue;
     top: 0;
-    z-index: 2;
+    z-index: 1;
```
Comment (xwatkins) [inline src/styles/components/data-table.scss:29]: I think there's some scss variables defined somewhere for `z-index`. It often gets messy so using the variables can help ensure where things are in relation to each other in the UI.

**Review by aurel-l — APPROVED** [review]
Given @xwatkins 's comments, once those are fixed I'd consider it approved from my side

File: src/styles/components/data-table.scss (hunk 2 above)
Comment (dlrice) [reply src/styles/components/data-table.scss]: good point. done

File: src/components/data-loader.jsx (hunk 1 above)
Comment (dlrice) [reply src/components/data-loader.jsx:56]: good point. done

**Pull request merged**